jsonschema/compat.py

```diff
@@ -40,6 +43,9 @@ def urlsplit(url):
     return SplitResult(scheme, netloc, path, query, fragment)
 
 
+DefragResult = namedtuple('DefragResult', 'url fragment')
```

Comment:
Ah OK, understood, up to you on whether to remove or not.

Basically I agree with all of the nice things you mentioned about why data objects are nice, but namedtuple's biggest issue is that it's... a `tuple`. That introduces all sorts of unintended consequences that people don't realize when they use it. Specifically to give just one example, it means that two namedtuples of *entirely different types* are comparable, and will compare equal if they contain the same positional elements. Really really really bad. I think you're right about namedtuple being convenient though, FWIW that's why something like `characteristic` (http://pypi.python.org/pypi/characteristic) is an attempt bridge the gap there.